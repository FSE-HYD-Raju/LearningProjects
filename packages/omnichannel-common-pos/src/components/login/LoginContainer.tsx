import * as React from "react";
import yup, { Schema } from "yup";
import { RouteComponentProps, withRouter } from "react-router";
import { SessionUtils, SessionKeys } from "../../utils/SessionUtils";
import { ContextType, contextTypesValidationMap } from "../../types";
import { User } from "../../redux";
import { commonDigitalLifeRoutes, commonServiceDeskRoutes } from "../../routes/commonRoutesMap";
import messages from "../../commonMessages";

interface LoginContainerActionProps {
	actions: {
		login: (email?: string, password?: string) => void;
		logout: () => void;
	};
}

interface LoginContainerStateProps {
	error?: any;
	user?: User;
}

interface LoginContainerOwnProps {
	alternativeRoute?: string;
	hideLogin?: () => void;
}

type LoginContainerProps = LoginContainerOwnProps & LoginContainerStateProps & LoginContainerActionProps & RouteComponentProps<any>;

interface LoginContainerState {
	email?: string;
	password?: string;
	error?: any;
	errors: object;
}

interface LoginSchemaModel {
	email?: string;
	password?: string;
}

interface LoginContainerProvidedProps extends LoginContainerState {
	handleInput: (loginInfo: {email: string, password: string}) => void;
	handleLogout: () => void;
	handleLogin: () => void;
	schema: Schema<LoginSchemaModel>;
	handleSubmit: () => void;
	onError: (errors: object) => void;
}

class LoginContainer extends React.Component<LoginContainerProps, LoginContainerState> {

	static contextTypes = contextTypesValidationMap;

	schema: Schema<LoginSchemaModel>;

	constructor(props: LoginContainerProps, context: ContextType) {
		super(props);

		const { formatMessage } = context.intl;

		this.schema = yup.object({
			email: yup.string().email(formatMessage(messages.loginEmailError)).required(formatMessage(messages.emailRequired)),
			password: yup.string().required(formatMessage(messages.passwordRequired))
		});
		this.state = {
			errors: {}
		};
	}

	handleInput = (loginInfo: LoginSchemaModel) => {
		this.setState({
			email: loginInfo.email,
			password: loginInfo.password,
			error: undefined,
			errors: {}
		});
	};

	handleSubmit = () => {
		const model: LoginSchemaModel = {
			email: this.state.email,
			password: this.state.password
		};
		this.schema.validate(model).then(() => {
			this.handleLogin();
		});
	};

	handleLogin = () => {
		this.props.actions.login(this.state.email, this.state.password);
	};

	handleLogout = () => {
		window.console.log("HANDLE LOGOUT");
		this.props.actions.logout();
	};

	onError = (errors: object) => {
		this.setState({ errors });
	};

	componentWillReceiveProps(newprops: LoginContainerProps) {
		if (newprops.error) {
			this.setState({ error: newprops.error });
		} else if (!this.props.user) {
			if (newprops.user && !newprops.error) {
				if (this.props.hideLogin) {
					this.props.hideLogin();
				}
				SessionUtils.removeItem(SessionKeys.activeBasketId);
				if (newprops.alternativeRoute) {
					this.props.history.push(newprops.alternativeRoute);
				} else {
					const app = process && process.env && process.env.omnichannel;
					switch (app) {
						case "telesales":
							this.props.history.push(commonServiceDeskRoutes.SERVICE_DESK_INDEX.createLink());
							break;
						case "pos":
							this.props.history.push(commonServiceDeskRoutes.SERVICE_DESK_INDEX.createLink());
							break;
						case "b2c":
							this.props.history.push(commonDigitalLifeRoutes.DIGITAL_LIFE_INDEX.createLink());
							break;
						case "cms_admin":
							this.props.history.push("/cmsadmin/overview/content");
							break;
						default:
							this.props.history.push(commonDigitalLifeRoutes.DIGITAL_LIFE_INDEX.createLink());
					}
				}
			}
		}
	}

	render() {
		return (
			<div className="LoginContainer">
				{
					React.Children.map(this.props.children, ((child: React.ReactNode) => {
						return React.cloneElement<LoginContainerProvidedProps>(child as React.ReactElement<LoginContainerProvidedProps>, {
							handleInput: this.handleInput,
							handleLogout: this.handleLogout,
							handleLogin: this.handleLogin,
							schema: this.schema,
							handleSubmit: this.handleSubmit,
							onError: this.onError,
							...this.state
						});
				}))}
			</div>
		);
	}
}

const LoginContainerWithRouter: React.ComponentClass<LoginContainerOwnProps & LoginContainerStateProps & LoginContainerActionProps> =
	withRouter(LoginContainer);

export {
	LoginContainer as LoginContainerUnwrapped,
	LoginContainerWithRouter as LoginContainer,
	LoginContainerProvidedProps,
	LoginContainerState,
	LoginContainerStateProps,
	LoginContainerOwnProps,
	LoginContainerProps,
	LoginContainerActionProps,
	LoginSchemaModel
};
