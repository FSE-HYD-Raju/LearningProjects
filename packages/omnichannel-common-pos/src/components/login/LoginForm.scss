.LoginForm {
	&-form-group {
		margin-bottom: 20px;
		> label {
			display: flex;
			justify-content: flex-start;
			align-items: center;

			@media screen and (max-width: 480px) {
				flex-direction: column;
				flex: 1;
				align-items: flex-start;
			}

			> span {
				min-width: 100px;
				font-size: 15px;
				font-weight: 400;

			}

			> div {
				@media screen and (max-width: 480px) {
					width: 100%;
				}
			}
		}
	}

	&-login-button {
		@media screen and (max-width: 480px) {
			width: 100%;
		}
	}

	&-login-button-container {
		display: flex;
		flex-direction: row;
		@media screen and (min-width: 480px) {
			justify-content: center;
		}
	}
}
