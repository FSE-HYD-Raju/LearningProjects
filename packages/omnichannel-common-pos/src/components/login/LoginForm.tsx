import ErrorLabel from "../../components/error/ErrorLabel";
import withFormal from "../ocComponents/withFormal";
import messages from "../../commonMessages";
import cssns from "../../utils/cssnsConfig";
import { ComponentType, FC } from "react";
import FormattedMessage from "../../channelUtils/FormattedMessage";
import { Schema } from "yup";
import { OcTextInput } from "../ocComponents/input/OcTextInput";
import { OcPasswordInput } from "../ocComponents/input/OcPasswordInput";
import { OcButton, OcButtonType } from "../ocComponents/button/OcButton";
const Form = require("react-formal");
const { React } = cssns("LoginForm");

const FormalOcInput: ComponentType<any> = withFormal(OcTextInput, {type: "text"});
const FormalPasswordInput: ComponentType<any> = withFormal(OcPasswordInput, {type: "password"});

interface LoginFormProps {
	id?: string;
	email?: string;
	password?: string;
	className?: string;
	fieldStyle?: {color: string};
	handleInput?: (loginInfo: {email: string, password: string}) => void;
	onError?: () => void;
	error?: any;
	errors?: object;
	noAccess?: boolean;
	handleSubmit?: (loginInfo: {email: string, password: string}) => void;
	schema?: Schema<{email: string, password: string}>;
}

const LoginForm: FC<LoginFormProps> = (props: LoginFormProps) => {
	const value = {
		email: props.email,
		password: props.password
	};

	const { fieldStyle } = props;

	if (props.schema) {
		return (
			<div className={`LoginForm ${props.className}`}>
				<Form
					schema={props.schema}
					onChange={props.handleInput}
					value={value}
					errors={props.errors}
					onError={props.onError}
				>
					<Form.Field
						label={<FormattedMessage{...messages.loginModalEmailInput}/>}
						name="email"
						type={FormalOcInput}
						style={fieldStyle}
						id={`${props.id}-LoginForm-email-field`}
						events={["onSubmit"]}
						labelColor={fieldStyle && fieldStyle.color}
					/>

					<Form.Field
						label={<FormattedMessage{...messages.loginModalPasswordInput}/>}
						name="password"
						type={FormalPasswordInput}
						style={fieldStyle}
						id={`${props.id}-LoginForm-password-field`}
						events={["onSubmit"]}
						labelColor={fieldStyle && fieldStyle.color}
					/>
					{(props.error || props.noAccess) && (
						<ErrorLabel message={<FormattedMessage {...messages.loginError} />}/>
					)}

					<div
						className="login-button-container"
						style={{ margin: "16px 0", paddingTop: "16px" }}
					>
						<OcButton
							onClick={props.handleSubmit as any}
							id={`${props.id}-LoginForm-submit-button`}
							buttonType={OcButtonType.PRIMARY}
							htmlBtnType="submit"
							block={true}
						>
							<FormattedMessage {...messages.loginButton} />
						</OcButton>
					</div>
				</Form>
			</div>
		);
	} else {
		return <span />;
	}
};

export {
	LoginFormProps,
	LoginForm
};
