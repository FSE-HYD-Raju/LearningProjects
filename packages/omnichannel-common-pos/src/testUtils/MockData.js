import _ from "lodash";
import R from "ramda";

const firstDateThisWeek = () => {
	const date = new Date();
	while (date.getDay() !== 0) {
		date.setDate(date.getDate() - 1);
	}
	return date;
};

const people = [
	{
		firstName: "Kathrin",
		lastName: "K",
		gender: "female",
		language: "swe",
		countryOfBirth: "SE",
		birthDay: firstDateThisWeek(),
		emails: [{ email: "kathrin.k@qvantel.com" }],
		email: "kathrin.k@qvantel.com",
		mobileNumbers: [{ number: "2398457" }],
		mobileNumber: "2398457",
		fixedLineNumbers: [{ number: "2398457" }],
		fixedLineNumber: "2398457",
		password: "qwerasdf",
		passwordConfirmation: "qwerasdf"
	},
	{
		firstName: "John",
		lastName: "Smith",
		gender: "male",
		language: "fin",
		birthDay: firstDateThisWeek(),
		countryOfBirth: "United Kingdom",
		emails: [{ email: "john.smith@qvantel.com" }],
		email: "john.smith@qvantel.com",
		mobileNumbers: [{ number: "3498573" }],
		mobileNumber: "3498573",
		fixedLineNumbers: [{ number: "52497584" }],
		fixedLineNumber: "52497584"
	}
];

const basketItems = [
	{
		id: "7879c247-9e98-40ee-bb48-544eb366494a",
		type: "basketItems",
		attributes: {
			product: {
				name: "Movie Steaming Service"
			},
			totalPrices: [
				{
					type: "ONE_TIME",
					taxFreeAmount: 1.9,
					taxRate: 0,
					currency: "EUR"
				},
				{
					type: "RECURRENT",
					taxFreeAmount: 14.9,
					taxRate: 0,
					currency: "EUR"
				}
			],
			childBasketItems: [],
			totalUpfrontPrice: 1.9,
			totalUpfrontPriceWithChildren: 1.9
		}
	},
	{
		id: "c0065ffc-77f4-4e44-b15c-fbd2f78d04ce",
		type: "basketItems",
		attributes: {
			product: {
				name: "XDSL"
			},
			totalPrices: [
				{
					type: "ONE_TIME",
					taxFreeAmount: 100.6,
					taxRate: 0,
					currency: "EUR"
				},
				{
					type: "RECURRENT",
					taxFreeAmount: 30,
					taxRate: 0,
					currency: "EUR"
				}
			],
			childBasketItems: []
		}
	}
];

const deliveryItem = {
	id: "c0065ffc-77f4-4e44-b15c-fbd2f78d04ce",
	type: "basketItems",
	inputtedCharacteristics: {
		CH_Delivery_Contact_Info: "5c5dddf6-72be-4d0a-9711-6027f247af28"
	},
	product: {
		name: "Home delivery",
		id: "PO_Home_Delivery"
	}
};

export default class MockData {
	static getPerson(searchCriteria) {
		if (searchCriteria) {
			return _.find(people, searchCriteria);
		} else {
			return people[0];
		}
	}

	static getPeople() {
		return people;
	}

	static getPostalAddress() {
		return {
			street: "1 Test St",
			coAddress: "Best Ltd.",
			postalCode: "90210",
			city: "Test City",
			country: "FI"
		};
	}

	static getUser() {
		return {
			"id": "bf588945-ce95-490a-a774-8ef91661a146",
			"type": "persons",
			"attributes": {
				"lastName": "Testeri",
				"addresses": null,
				"gender": "male",
				"fixedLineNumbers": [],
				"formattedName": "Teppo Testeri",
				"roles": null,
				"billingAccountId": null,
				"language": null,
				"emails": [
					{
						"role": "PRIMARY",
						"id": "2ab79d7b-862d-49df-8afc-e6104a914919",
						"email": "tepsteri@tepinposti.fi"
					}
				],
				"honorificSuffix": null,
				"countryOfBirth": "FI",
				"personRelationships": [],
				"birthDay": null,
				"placeOfBirth": null,
				"characteristics": null,
				"mobileNumbers": [
					{
						"role": "PRIMARY",
						"id": "a843f46d-f729-40dd-a7ce-240a64bc008e",
						"number": "1234576890"
					}
				],
				"honorificPrefix": null,
				"avatar": "/img/icon-user-male.svg",
				"privacySettings": {
					"privacy1": false,
					"privacy2": false
				},
				"postalAddresses": [
					{
						"role": "PRIMARY",
						"id": "5e30221d-ad19-4043-8088-f594f5fdf77f",
						"street": "Piippukatu 11 A 1",
						"coAddress": null,
						"postalCode": "40100",
						"city": "Roma",
						"country": "FI",
						"county": null,
						"description": null,
						"postOfficeBox": null,
						"stateOrProvince": null,
						"addressRegisterId": null,
						"apartment": null,
						"building": null
					},
					{
						"role": "DELIVERY",
						"id": "5c5dddf6-72be-4d0a-9711-6027f247af28",
						"street": "Piippukatu 11 A 1",
						"coAddress": "Piippukatu 11 A 1",
						"postalCode": "40100",
						"city": "Roma",
						"country": "FI",
						"county": null,
						"description": null,
						"postOfficeBox": null,
						"stateOrProvince": "",
						"addressRegisterId": null,
						"apartment": null,
						"building": null
					}
				],
				"identifications": [],
				"firstName": "Teppo",
				"customerAccountId": null,
				"nationality": null,
				"additionalName": null,
				"maritalStatus": null
			}
		}
	}

	static getBasket(withItems: boolean, withDeliveryItems: boolean) {
		const basket = {
			activeBasket: {
				attributes: {
					totalPrices: [],
					totalUpfrontPrice: 0
				}
			}
		};

		if (withItems) {
			basket.basketItems = basketItems;

			const totalPriceOneTime = R.reduce(
				R.add,
				0,
				basketItems.map(bi => {
					return bi.attributes.totalPrices[0].taxFreeAmount;
				})
			);
			const totalPriceMonthly = R.reduce(
				R.add,
				0,
				basketItems.map(bi => {
					return bi.attributes.totalPrices[0].taxFreeAmount;
				})
			);

			basket.activeBasket.attributes.totalPrices = [
				{
					type: "ONE_TIME",
					taxFreeAmount: totalPriceOneTime,
					taxRate: 0,
					currency: "EUR"
				},
				{
					type: "RECURRENT",
					taxFreeAmount: totalPriceMonthly,
					taxRate: 0,
					currency: "EUR"
				}
			];

			if (withDeliveryItems) {
				basket.basketItems[0].attributes.childBasketItems.push(
					deliveryItem
				);
			}
		}

		return basket;
	}
}
