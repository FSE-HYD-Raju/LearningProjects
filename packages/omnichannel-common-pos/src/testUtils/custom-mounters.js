import React from "react";
import { IntlProvider, intlShape } from "react-intl";
import { mount, shallow } from "enzyme";
import PropTypes from "prop-types";
import R from "ramda";
import TestUtils from "./TestUtils";

const MAX_DIVE_DEEP_LEVEL = 16;
const intlProvider = new IntlProvider({ locale: "en" }, {});
const { intl } = intlProvider.getChildContext();

const baseContext = {
	flux: {
		persist: () => {}
	},
	store: TestUtils.mockReduxStore(),
	intl,
	location: {},
	only: {},
	skip: {}
};

const nodeWithIntlProp = (node, mockedRouterOptions) => {
	if (mockedRouterOptions) {
		mockedRouterOptions.props().history.push = jest.fn();
	}
	return React.cloneElement(node, {
		intl,
		...(mockedRouterOptions ? mockedRouterOptions.props() : {})
	});
};

function getOptionsForShallowMount(options, debug, mockedRouterContextOptions) {
	const mountOptions = R.merge(options ? options : {}, {
		disableLifecycleMethods: !(
			options && options.disableLifecycleMethods === false
		),
		context: R.mergeAll([
			debug
				? {
						...baseContext
					}
				: baseContext,
			R.omit("intl", options && options.context ? options.context : {}),
			mockedRouterContextOptions.context
		]),
		childContextTypes: R.merge(
			{
				intl: intlShape,
				store: PropTypes.object
			},
			mockedRouterContextOptions.childContextTypes
		)
	});
	TestUtils.mockReduxStoreToFlux(mountOptions.context);
	return mountOptions;
}
function deepDive(wrapper) {
	let diveAttemptsCount = 0;
	while (
		wrapper.hostNodes().length === 0 &&
		diveAttemptsCount < MAX_DIVE_DEEP_LEVEL
	) {
		wrapper = wrapper.dive();
		diveAttemptsCount += 1;
	}
	return wrapper;
}
export function shallowWithContextAndRouterProps(node, options, debug) {
	return shallowWithContext(node, options, debug, true);
}
export function shallowWithContextAndLifecycle(node, options, debug) {
	return shallowWithContext(
		node,
		R.merge(options || {}, { disableLifecycleMethods: false }),
		debug,
		true
	);
}
export function deepShallowWithContextAndLifecycle(node, options, debug) {
	return deepDive(shallowWithContextAndLifecycle(node, options, debug));
}
export function deepShallowWithContext(node, options, debug) {
	return deepDive(shallowWithContext(node, options, debug, true));
}
export function shallowWithContext(node, options, debug, injectRouterProps) {
	const mockedRouterOptions = TestUtils.getMockedRouterOptions();
	const mockedRouterContextOptions = mockedRouterOptions.get();
	return shallow(
		nodeWithIntlProp(node, injectRouterProps ? mockedRouterOptions : null),
		getOptionsForShallowMount(options, debug, mockedRouterContextOptions)
	);
}

export function mountWithContextAndRouterProps(node, options, debug) {
	return mountWithContext(node, options, debug, true);
}
export function mountWithContext(node, options, debug, injectRouterProps) {
	const mockedRouterOptions = TestUtils.getMockedRouterOptions();
	const mockedRouterContextOptions = mockedRouterOptions.get();
	const mountOptions = R.merge(options ? options : {}, {
		context: R.mergeAll([
			debug
				? {
						...baseContext
					}
				: baseContext,
			R.omit("intl", options && options.context ? options.context : {}),
			mockedRouterContextOptions.context
		]),
		childContextTypes: R.merge(
			{
				flux: PropTypes.object,
				store: PropTypes.object,
				intl: intlShape,
				location: PropTypes.object,
				only: PropTypes.object,
				skip: PropTypes.object
			},
			mockedRouterContextOptions.childContextTypes
		)
	});
	TestUtils.mockReduxStoreToFlux(mountOptions.context);
	return mount(
		nodeWithIntlProp(node, injectRouterProps ? mockedRouterOptions : null),
		mountOptions
	);
}

const attachmentElementId = "__OC_COMPONENT_TEST_ATTACHMENT_ELEMENT__";

export function attachWithContext(node, options) {
	const attachmentElement = document.createElement("div");
	attachmentElement.setAttribute("id", attachmentElementId);
	document.body.appendChild(attachmentElement);

	const wrapper = mountWithContext(
		node,
		R.merge(options, { attachTo: attachmentElement })
	);
	const doppelWrapper = hijackDetacher(wrapper);

	return doppelWrapper;
}

function hijackDetacher(wrapper) {
	const doppel = wrapper;
	doppel.__detach = wrapper.detach;

	doppel.detach = () => {
		removeAttachmentElements();
		doppel.__detach.apply(wrapper);
	};

	return doppel;
}

function removeAttachmentElements() {
	document.body.querySelectorAll("#" + attachmentElementId).forEach(elem => {
		document.body.removeChild(elem);
	});
}

export function detach(wrapper) {
	removeAttachmentElements();
	wrapper.detach();
}
