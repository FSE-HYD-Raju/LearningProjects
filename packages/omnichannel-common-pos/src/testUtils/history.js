import isClient from "../utils/isClient";
import { createBrowserHistory, createMemoryHistory } from "history";
import { useRouterHistory } from "react-router";

const history = isClient
	? useRouterHistory(createBrowserHistory)({ queryKey: false })
	: createMemoryHistory();

export default history;
