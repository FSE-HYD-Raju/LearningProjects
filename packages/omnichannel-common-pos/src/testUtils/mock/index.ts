import { default as productOffering } from "./ProductOfferingMock";
import { ProductMock as product } from "./ProductMock";
import { AgreementMock as agreement } from "./AgreementMock";
import { ChargingBalancesMock as chargingBalances } from "./ChargingBalancesMock";
import { default as postalAddress } from "./PostalAddressMock";
import { default as identification } from "./IdentificationMock";
import { default as user } from "./UserMock";
import { default as order } from "./OrderMock";
import { default as basket } from "./BasketMock";
export { productOffering, postalAddress, identification, user, order, basket, product, agreement, chargingBalances };
