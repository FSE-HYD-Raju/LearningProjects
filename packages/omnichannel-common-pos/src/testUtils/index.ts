import ReactWidgetsTestUtils from "./ReactWidgetsTestUtils";
import * as MockDataMaker from "./mock";
const {
	shallowWithContextAndRouterProps,
	shallowWithContextAndLifecycle,
	deepShallowWithContextAndLifecycle,
	deepShallowWithContext,
	shallowWithContext,
	mountWithContextAndRouterProps,
	mountWithContext,
	attachWithContext,
	detach
} = require("./custom-mounters");
const TestUtils = require("./TestUtils");
const MockData = require("./MockData");
const SimpleDataMock = require("./mock/simpleDataMock");

export {
	MockData,
	ReactWidgetsTestUtils,
	TestUtils,
	SimpleDataMock,
	shallowWithContextAndRouterProps,
	shallowWithContextAndLifecycle,
	deepShallowWithContextAndLifecycle,
	deepShallowWithContext,
	shallowWithContext,
	mountWithContextAndRouterProps,
	mountWithContext,
	attachWithContext,
	detach,
	MockDataMaker
};

export * from "./MockRouterProps";

export default TestUtils;
