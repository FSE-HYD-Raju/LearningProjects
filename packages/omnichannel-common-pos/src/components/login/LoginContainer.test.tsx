import * as React from "react";
import { shallowWithContext, mountWithContext } from "../../testUtils";
import { LoginContainerUnwrapped as LoginContainer, LoginContainerProps } from "./LoginContainer";
import { LoginForm } from "./LoginForm";
import { mockRouterProps } from "../../testUtils";

const minProps: LoginContainerProps = {
	...mockRouterProps,
	actions: {
		login: jest.fn(),
		logout: jest.fn(),
	}
};

describe("LoginContainer", () => {
	it("succeeds at shallow mount with minimal props", () => {
		const wrapper = shallowWithContext(<LoginContainer {...minProps} />);
		expect(wrapper).toMatchSnapshot();
	});

	it("succeeds at deep mount without props", () => {
		mountWithContext(<LoginContainer  {...minProps}/>);
	});

	it("should render LoginForm as child", () => {
		const wrapper = mountWithContext(
			<LoginContainer {...minProps}>
				<LoginForm id="LoginDropdown" />
			</LoginContainer>
		);

		const loginContainer = wrapper.find(".LoginContainer");
		expect(loginContainer.length).toEqual(1);

		const loginForm = loginContainer.find(".LoginForm");
		expect(loginForm.length).toEqual(1);
	});

	it("should pass all props to child", () => {
		class MyChild extends React.Component<any> {
			render() {
				return <div />;
			}
		}

		const props = {
			...minProps,
			children: <MyChild />
		};
		const wrapper = mountWithContext(<LoginContainer {...props} />);

		const loginContainer = wrapper.find("LoginContainer");
		const myChild = loginContainer.find(MyChild);
		const childProps = Object.getOwnPropertyNames(myChild.props());

		expect(childProps).toContain("handleInput");
		expect(childProps).toContain("handleLogout");
		expect(childProps).toContain("handleLogin");
		expect(childProps).toContain("schema");
		expect(childProps).toContain("handleSubmit");
		expect(childProps).toContain("onError");
	});
});
