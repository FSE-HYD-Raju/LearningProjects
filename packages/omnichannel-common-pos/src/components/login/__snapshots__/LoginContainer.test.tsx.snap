// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginContainer succeeds at shallow mount with minimal props 1`] = `
<div
  className="LoginContainer"
/>
`;
